src/System.Private.CoreLib/shared/System/Number.Formatting.cs

```diff
@@ -1607,6 +1607,7 @@ internal static unsafe char ParseFormatSpecifier(ReadOnlySpan<char> format, out
         internal static unsafe void NumberToString(ref ValueStringBuilder sb, ref NumberBuffer number, char format, int nMaxDigits, NumberFormatInfo info)
         {
             number.CheckConsistency();
+            bool isCorrectlyRounded = (number.Kind == NumberBufferKind.FloatingPoint);
```

Comment:
Standard numeric-format strings go through here and are always correctly rounded for floating-point values.